lib/matplotlib/tests/test_legend.py

```diff
@@ -974,7 +974,6 @@ def test_legend_pathcollection_labelcolor_markfacecolor_cmap():
         np.arange(10),
         np.arange(10),
         label='#1',
-        c=np.arange(10),
```

Comment:
I think by removing `c` here the test becomes the same as the one above it.  Instead, since we expect the warning here we can use the [pytest.warns](https://docs.pytest.org/en/stable/how-to/capture-warnings.html#assertwarnings) context manager.

Reply:
I was considering changing the assertion to .raises(). However, that would completely change the intent of the test. We should rename the test in that case. I would love if @chahak13, the original author of the test, would chime in and tell us what we should do.

Reply:
The correct fix is to remove `facecolors=facecolors`. It has no effect in the test because `c` takes precedence.

Nevertheless, we should have an extra tests that exercises the warning.

Reply:
Good point, @timhoffm. I will change the test to use c to receive the colormap and add a test for the warning.